server/rest/src/main/java/org/infinispan/rest/operations/CacheOperations.java

```diff
@@ -206,21 +206,23 @@ public InfinispanResponse deleteCacheValue(InfinispanRequest request) throws Res
 
          if (entry instanceof InternalCacheEntry) {
             InternalCacheEntry<String, Object> ice = (InternalCacheEntry<String, Object>) entry;
-            Metadata meta = entry.getMetadata();
-            if (meta instanceof MimeMetadata) {
-               String etag = calcETAG(ice.getValue(), ((MimeMetadata) meta).contentType());
-               Optional<String> clientEtag = request.getEtagIfNoneMatch();
-               if (clientEtag.map(t -> t.equals(etag)).orElse(true)) {
-                  response.status(HttpResponseStatus.OK);
-                  if (useAsync.isPresent() && useAsync.get()) {
-                     restCacheManager.getCache(cacheName).removeAsync(key);
+            Optional<String> clientEtag = request.getEtagIfNoneMatch();
+            if (clientEtag.isPresent()) {
+               Metadata meta = entry.getMetadata();
+               if (meta instanceof MimeMetadata) {
+                  String etag = calcETAG(ice.getValue(), ((MimeMetadata) meta).contentType());
+
+                  if (clientEtag.map(t -> t.equals(etag)).orElse(true)) {
```

Comment:
We're in the `clientEtag.isPresent()` branch, so this is equivalent to `clientEtag.get().equals(etag)`

Reply:
Fixed